src/main.rs

```diff
@@ -38,31 +38,36 @@ struct MarketData {
 
 impl MarketData {
     fn insert_and_validate(&mut self, hash: &str, filerequest: &FileRequest) {
-        // let mut holders: Vec<FileRequest> = vec![];
         // check if self.files[hash] exists
         if !self.files.contains_key(hash) {
-            self.files.insert(hash.to_string(), vec![filerequest.clone()]);
+            self.files
+                .insert(hash.to_string(), vec![filerequest.clone()]);
         }
         let current_time = get_current_time();
-        let mut holder_indices = vec![];
-        for (pos, holder) in self.files[hash].iter().enumerate() {
+        let producers = self.files.get_mut(hash).unwrap(); // safe to unwrap since we already checked if the key exists
```

Comment:
Suggestion (non-blocking): since this is safe to unwrap, you can just index normally like this
```Rust
let producers = &self.files[hash]
```
since it'll have the same effect.

Reply:
I tried using that earlier but the compiler recommended using get_mut instead since im changing the value

Reply:
Replaced with match